microcosm_eventsource/func.py

```diff
@@ -0,0 +1,67 @@
+"""
+Custom database functions.
+
+"""
+from sqlalchemy.sql.expression import FunctionElement
+from sqlalchemy.ext.compiler import compiles
+
+
+class last(FunctionElement):
+    """
+    Define a SQLAlchemy function that maps to a postgres function to select the last non-null value in window.
+
+    """
+    name = "last"
+
+    @staticmethod
+    def create(connection):
+        """
+        Create the postgres functions for this implementation.
+
+        XXX Not currently called in any automatic way.
+
+        """
+        connection.execute("""
```

Comment:
I recommend searching for "postgres create function" and "postgres create aggregate"